cant compile xaml file

### Current behavior

I am getting this error on compilation

D:\source\ZiiDMS\ZiiDMSApp\Client\ZiiDMSApp\ZiiDMSApp\Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.Xam
lCodeGenerator\ZtShellPage_1417a2ab143b14d67a03ae0ce9ec7b98.cs(221,264): error CS0118: '___tctx' is a variable but is u
sed like a type [D:\source\ZiiDMS\ZiiDMSApp\Client\ZiiDMSApp\ZiiDMSApp\ZiiDMSApp.csproj::TargetFramework=net8.0-android
]

I try to look and see what file its generating but i cant find it.  This is happening with all platforms.

Any idea of what i am doing wrong? something wrong with my nugets?


### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

--- a/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.cs
+++ b/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.cs
@@ -157,7 +157,7 @@ namespace Uno.UI.SourceGenerators.XamlGenerator.Utils
 				throw new Exception("Unexpected XBindRoot");
 			}
 
-			public void BuildPath(XBindPath path, StringBuilder? propertyBuilder)
+			public void BuildPath(XBindPath path, StringBuilder? propertyBuilder, bool isType = false)
 			{
 				if (path is XBindMemberAccess memberAccess)
 				{
@@ -176,26 +176,28 @@ namespace Uno.UI.SourceGenerators.XamlGenerator.Utils
 				}
 				else if (path is XBindIdentifier identifier)
 				{
-					if (!identifier.IdentifierText.StartsWith("global::", StringComparison.Ordinal) &&
+					if (!isType &&
+						!identifier.IdentifierText.StartsWith("global::", StringComparison.Ordinal) &&
 						identifier.IdentifierText is not ("null" or "true" or "false"))
 					{
 						_builder.Append(_contextName);
 						_builder.Append('.');
 					}
 
-					_builder.Append(identifier.IdentifierText);
+					if (isType)
+					{
+						_builder.Append(GetGlobalizedTypeName(identifier.IdentifierText));
+					}
+					else
+					{
+						_builder.Append(identifier.IdentifierText);
+					}
+
 					propertyBuilder?.Append(identifier.IdentifierText);
 				}
 				else if (path is XBindAttachedPropertyAccess attachedPropertyAccess)
 				{
-					if (attachedPropertyAccess.PropertyClass is XBindIdentifier attachedClassIdentifier)
-					{
-						_builder.Append(GetGlobalizedTypeName(attachedClassIdentifier.IdentifierText));
-					}
-					else
-					{
-						BuildPath(attachedPropertyAccess.PropertyClass, propertyBuilder: null);
-					}
+					BuildPath(attachedPropertyAccess.PropertyClass, propertyBuilder: null, isType: true);
 
 					_builder.Append(".Get");
 					_builder.Append(attachedPropertyAccess.PropertyName.IdentifierText);
@@ -217,7 +219,8 @@ namespace Uno.UI.SourceGenerators.XamlGenerator.Utils
 				else if (path is XBindCast xBindCast)
 				{
 					_builder.Append('(');
-					BuildPath(xBindCast.Type, propertyBuilder: null);
+					BuildPath(xBindCast.Type, propertyBuilder: null, isType: true);
+
 					_builder.Append(')');
 					BuildPath(xBindCast.Expression, propertyBuilder);
 				}

--- a/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Controls/When_xBind_With_Cast_Default_Namespace.xaml
+++ b/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Controls/When_xBind_With_Cast_Default_Namespace.xaml
@@ -1,0 +1,13 @@
+﻿<Page
+    x:Class="Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls.When_xBind_With_Cast_Default_Namespace"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls"
+	xmlns:muxc="using:Microsoft.UI.Xaml.Controls"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d"
+    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+
+	<TextBlock x:Name="tb" x:FieldModifier="public" Text="{x:Bind ((ContentControl)ViewModel.P).Content}" />
+</Page>

--- a/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Controls/When_xBind_With_Cast_Default_Namespace.xaml.cs
+++ b/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Controls/When_xBind_With_Cast_Default_Namespace.xaml.cs
@@ -1,0 +1,20 @@
+﻿using System.ComponentModel;
+using Microsoft.UI.Xaml.Controls;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls;
+
+public sealed partial class When_xBind_With_Cast_Default_Namespace : Page
+{
+	public ViewModel ViewModel { get; } = new();
+
+	public When_xBind_With_Cast_Default_Namespace()
+	{
+		this.DataContext = ViewModel;
+		this.InitializeComponent();
+	}
+}
+
+public class ViewModel
+{
+	public ContentControl P => new ContentControl() { Content = "Hello" };
+}

--- a/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_xBind.cs
+++ b/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_xBind.cs
@@ -29,4 +29,14 @@ public class Given_xBind
 		SUT.ItemOther.IsSelected = true;
 		Assert.AreEqual("ItemOther", SUT.tb.Text);
 	}
+
+	[TestMethod]
+	public async Task When_xBind_With_Cast_Default_Namespace()
+	{
+		var SUT = new When_xBind_With_Cast_Default_Namespace();
+		TestServices.WindowHelper.WindowContent = SUT;
+		await TestServices.WindowHelper.WaitForLoaded(SUT);
+
+		Assert.AreEqual("Hello", SUT.tb.Text);
+	}
 }

